### PR TITLE
docs: remove docs for properties that have been removed, fix most warnings from docgen

### PR DESCRIPTION
**Pull request by ewanharris**
**JIRA:** https://jira.appcelerator.org/browse/TIDOC-3289

This PR fixes up the docs to do the following, it would be great if the initial authors of PRs associated to the changes/doc authors can review to make sure I haven't screwed up 🙂 I've tried to split the commits up so that it's a single commit per reviewable change

@garymathews Could you please take a look at the following
637312e262b062d0f9deef0a82d1628d9d199ccd - Update Geolocation docs in accordance with #10547. Keep an eye out for `// check me!` as there are still references to the removed properties (and `legacy mode`) that I'm unsure as to what to do with

@vijaysingh-axway Could you please take a look over the following
d77316f50793216fa31662e033be35cc068402c5 - Fix reference to lineBreakMode
7281537c5b1c0e5fa37c0c01097bcaa1bbd18a93 / a2ec4ac63630d4d3df223237aa6f42d297ed3359 - Fix references in webview docs

@jawa9000  Could you please take a look over the following (and the others if you have time please)
2b97935687af4bd643406cde975f0dabfd9a4369 -  Correct name of contentOffset

Free for all commits 😆
7e25d1adce7d38fc16ab84bed702ba9a25314c8a - Fix typos
14ea75727d0a7bd3ad70fdd083d9ff05a997e1d9 - Adds API that referenced a class, makes doing this work much easier to know the full string to look for
feb7bbbb3bba9382acfe5234a9992801c52b8407 - Pass methods into the getter/setter generation to avoid creating invalid getters/setters

### CODE REVIEW

File: apidoc/Titanium/Geolocation/Geolocation.yml [hunk 1]
```diff
@@ -612,50 +594,6 @@ properties:
     permission: read-only
     platforms: [iphone, ipad]
 
-  - name: PROVIDER_GPS
```
Comment (garymathews) [inline apidoc/Titanium/Geolocation/Geolocation.yml]: These providers are still present and can be used if `ti.playservices` is not included.

**Review by garymathews — CHANGES_REQUESTED** [review]

File: apidoc/Titanium/Android/Notification.yml [hunk 2]
```diff
@@ -358,7 +358,7 @@ properties:
         ([Filesystem.externalStorageDirectory](Titanium.Filesystem.externalStorageDirectory)).
 
         Note that this sound property is ignored on Android 8.0 and above. On these OS versions,
-        you must use the [NotificationChannel.sound](Titanium.Android.Titanium.Android.NotificationChannel.sound)
+        you must use the [NotificationChannel.sound]Titanium.Android.NotificationChannel.sound)
```
Comment (sgtcoolguy) [inline apidoc/Titanium/Android/Notification.yml]: Both before and after here look incorrect to me...

Is this correct?
```suggestion
        you must use the [NotificationChannel.sound](Titanium.Android.NotificationChannel.sound)
```

File: apidoc/Titanium/UI/iOS/iOS.yml [hunk 3]
```diff
@@ -1955,23 +1955,23 @@ properties:
   - name: CREDENTIAL_PERSISTENCE_NONE
     summary: Credential should not be stored.
     description: |
-        Use with <Titanium.UI.WebView.basicAuthentication.persistence>
+        Used with the `persistence` property of <Titanium.UI.WebView.setBasicAuthentication> to set the persistence level.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/iOS/iOS.yml]: It's an argument, not a property.

File: apidoc/Titanium/UI/iOS/iOS.yml [hunk 4]
```diff
@@ -1955,23 +1955,23 @@ properties:
...
     type: Number
     since: "8.0.0"
     permission: read-only
 
   - name: CREDENTIAL_PERSISTENCE_FOR_SESSION
     summary: Credential should be stored only for this session.
     description: |
-        Use with <Titanium.UI.WebView.basicAuthentication.persistence>
+        Used with the `persistence` property of <Titanium.UI.WebView.setBasicAuthentication> to set the persistence level.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/iOS/iOS.yml]: arg, not prop

File: apidoc/Titanium/UI/iOS/iOS.yml [hunk 5]
```diff
@@ -1955,23 +1955,23 @@ properties:
...
     type: Number
     since: "8.0.0"
     permission: read-only
 
   - name: CREDENTIAL_PERSISTENCE_PERMANENT
     summary: Credential should be stored in the keychain.
     description: |
-        Use with <Titanium.UI.WebView.basicAuthentication.persistence>
+       Used with the `persistence` property of <Titanium.UI.WebView.setBasicAuthentication> to set the persistence level.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/iOS/iOS.yml]: same as above

File: apidoc/Titanium/UI/iOS/iOS.yml [hunk 6]
```diff
@@ -1981,7 +1981,7 @@ properties:
         Credential should be stored permanently in the keychain, and in addition should be 
         distributed to other devices based on the owning AppleID.
     description: |
-        Use with <Titanium.UI.WebView.basicAuthentication.persistence>
+        Used with the `persistence` property of <Titanium.UI.WebView.setBasicAuthentication> to set the persistence level.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/iOS/iOS.yml]: same as above

File: apidoc/Titanium/UI/iOS/iOS.yml [hunk 7]
```diff
@@ -2098,7 +2098,7 @@ properties:
     summary: |
         Inject the script after the document element is created, but before any other content is loaded.
     description: |
-        Use with <Titanium.UI.WebView.addUserScript.injectionTime>.
+        Used to set the `injectionTime` property of <Titanium.UI.WebView.addUserScript>.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/iOS/iOS.yml]: arg, not prop

File: apidoc/Titanium/UI/iOS/iOS.yml [hunk 8]
```diff
@@ -2107,7 +2107,7 @@ properties:
     summary: |
         Inject the script after the document finishes loading, but before other subresources finish loading.
     description: |
-        Use with <Titanium.UI.WebView.addUserScript.injectionTime>.
+        Used to set the `injectionTime` property of <Titanium.UI.WebView.addUserScript>.
```
Comment (sgtcoolguy) [inline apidoc/Titanium/UI/iOS/iOS.yml]: arg, not prop

**Review by sgtcoolguy — CHANGES_REQUESTED** [review]

**Review by sgtcoolguy — APPROVED** [review]

**Pull request merged**